tests/test_api_callable.py

```diff
@@ -141,6 +141,27 @@ def test_call_kwargs(self):
         self.assertEqual(my_callable(None, CallOptions(key='updated')),
                          'updated')
 
+    def test_call_merge_options_metadata(self):
+        settings_kwargs = {
+            'key': 'value',
+            'metadata': [('key1', 'val1'), ('key2', 'val2')]
+        }
+
+        settings = _CallSettings(kwargs=settings_kwargs)
+        my_callable = api_callable.create_api_call(
+            lambda _req, _timeout, **kwargs: kwargs, settings)
+
+        expected_kwargs = settings_kwargs
+        self.assertEqual(my_callable(None), expected_kwargs)
+
+        expected_kwargs = {
+            'key': 'value',
+            'metadata': [('key1', 'updated_val1'), ('key2', 'val2')]
+        }
+        self.assertEqual(
+            my_callable(None, CallOptions(metadata=[('key1', 'updated_val1')])),
```

Comment:
Sorry, my mistake. What actually is happening is

- metadata from CallOptions goes first
- metadata from CallSettings goes after, except keys which are already present in CallOptions

> This is the key point of this PR: merging not the dict itself (it is already done in merge() method), but deeply merging *only one* specific item in the dict

That's not what the code in this PR does though?